File: Backend/Models/TestRequest.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class TestRequest
{
    [Key]
    public int TestID { get; set; }
    public int? RecordID { get; set; }
    public int? PatientID { get; set; }
    public string? TestType { get; set; }
    public string? Description { get; set; }
    public DateTime? RequestDate { get; set; }
    public DateTime? ResultDate { get; set; }
    public string? Result { get; set; }
    public string? Status { get; set; }
    public decimal? Fee { get; set; }

    // Navigation properties
    public virtual MedicalRecord? MedicalRecord { get; set; }
    public virtual Patient? Patient { get; set; }
}

File: SpringBoot/src/main/java/com/hospital/qlbv/entity/TestRequest.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "test_requests")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TestRequest {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer testId;
    
    @Column(name = "record_id")
    private Integer recordId;
    
    @Column(name = "patient_id")
    private Integer patientId;
    
    @Column(name = "test_type")
    private String testType;
    
    private String description;
    
    @Column(name = "request_date")
    private String requestDate;
    
    @Column(name = "result_date")
    private String resultDate;
    
    private String result;
    private String status;
    private Double fee;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "record_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"patient", "doctor", "prescriptions", "testRequests"})
    private MedicalRecord medicalRecord;
}
